src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -432,25 +431,42 @@ public void controlProperties(List<ControllableProperty> controllableProperties)
 		final List<AdvancedControllableProperty> advancedControllableProperties = new ArrayList<>();
 		final StringBuilder errorMessages = new StringBuilder();
 
-		tryParseIntAdapterProperties(errorMessages);
-		checkOutOfRange(errorMessages);
-
-		if (errorMessages.toString().length() > 0) {
-			throw new IllegalArgumentException(errorMessages.toString());
+		if (sequenceNumber == Integer.MAX_VALUE - Command.values().length) {
+			// If sequence number reach the max value of integer before all commands in this adapter will be executed-> Reset to default value
+			sequenceNumber = 0;
 		}
 
 		if (restCommunicator == null) {
 			try {
 				initAverRestCommunicator();
+			} catch (Exception e) {
+				if (this.logger.isErrorEnabled()) {
+					this.logger.error("error: Cannot init Rest communicator: " + this.host + " port: " + this.port);
+				}
+				restCommunicator.destroy();
```

Comment:
do you need to call destroy() here? It is normally not adapter's responsibility to do it, why is it necessary here?

Reply:
discussed offline already, thank you